app/Community/Controllers/MessageController.php

```diff
@@ -38,9 +38,20 @@ public function store(MessageRequest $request): RedirectResponse
                 return back()->withErrors(__('legacy.error.error'));
             }
 
+            foreach ($thread->users as $threadUser) {
+                if (!$threadUser->is($user) && !$user->can('sendToRecipient', [Message::class, $threadUser])) {
+                    return back()->withErrors(__('legacy.error.permissions'));
+                }
+            }
+
             (new AddToMessageThreadAction())->execute($thread, $user, $input['body']);
         } else {
             $recipient = User::firstWhere('User', $input['recipient']);
+
+            if (!$user->can('createForRecipient', [MessageThread::class, $recipient])) {
+                return back()->withErrors(__('legacy.error.permissions'));
+            }
```

Comment:
Do we really need two separate policies for this? Should this also just check `$user->can('sendToRecipient', [Message::class, $recipient]`?

Reply:
Reduced down to `MessagePolicy::sendToRecipient()` in latest.

Reply:
Does `MessageThread::class` need to be changed to `Message::class`?

Reply:
Yes, thanks. Updated in latest.